src/libraries/Common/src/System/Buffers/ArrayBufferWriter.cs

```diff
@@ -81,15 +81,41 @@ public ArrayBufferWriter(int initialCapacity)
         /// Clears the data written to the underlying buffer.
         /// </summary>
         /// <remarks>
-        /// You must clear the <see cref="ArrayBufferWriter{T}"/> before trying to re-use it.
+        /// You must reset or clear the <see cref="ArrayBufferWriter{T}"/> before trying to re-use it.
         /// </remarks>
+        /// <remarks>
+        /// Using <seealso cref="ResetWrittenCount"/> method is more performance effective.
+        /// </remarks>
+        /// <remarks>
+        /// If you clear the buffer data by <see cref="Clear"/> method, <see cref="GetMemory"/> method will return clean <see cref="Memory{T}"/>.
+        /// </remarks>
+        /// <seealso cref="ResetWrittenCount"/>
         public void Clear()
         {
             Debug.Assert(_buffer.Length >= _index);
             _buffer.AsSpan(0, _index).Clear();
             _index = 0;
         }
 
+        /// <summary>
+        /// Resets the data written to the underlying buffer without zeroing memory.
+        /// </summary>
+        /// <remarks>
+        /// You must reset or clear the <see cref="ArrayBufferWriter{T}"/> before trying to re-use it.
+        /// </remarks>
+        /// <remarks>
+        /// If you reset the buffer data by <see cref="ResetWrittenCount"/> method, <see cref="GetMemory"/> method may return non-clean <see cref="Memory{T}"/>.
+        /// </remarks>
+        /// <remarks>
+        /// To clear buffer data after <see cref="ResetWrittenCount"/> method next code can be used in user code:
+        /// <code>
+        /// buffer.Advance(buffer.FreeCapacity);
+        /// buffer.Clear();
+        /// </code>
+        /// </remarks>
+        /// <seealso cref="Clear"/>
+        public void ResetWrittenCount() => _index = 0; // just reset index
```

Comment:
nit: the code is self-describing, we don't need a comment for it

```suggestion
        public void ResetWrittenCount() => _index = 0;
```